index.bs

```diff
@@ -1798,6 +1822,66 @@ Note: Invoking this method from a [=browsing context=] where the [=Web Authentic
 
 </div>
 
+### Client Supported Capabilities (dictionary <dfn dictionary>SupportedCapabilities</dfn>) ### {#dictionary-supportedcapabilities}
+
+<xmp class="idl">
+    dictionary SupportedCapabilities {
+        boolean areU2FDevicesSupported;
+        boolean areCTAPDevicesSupported;
+        boolean areResidentKeysSupported;
+        boolean isUserVerificationSupported;
+        boolean isTokenBindingSupported;
+        UserVerifyingPlatformAuthenticatorAvailability userVerifyingPlatformAuthenticatorAvailability;
+        sequence<AuthenticatorTransport> transportsSupported;
+        sequence<DOMString> extensionsSupported;
+    };
+</xmp>
+
+This dictionary is used by [=[WRPS]=] to determine the supported capabilities of the [=client=].
+
+<div dfn-type="dict-member" dfn-for="SupportedCapabilities">
+    :   <dfn>areU2FDevicesSupported</dfn>
+    ::  This attribute indicates whether the [=client=] supports communicating with [[FIDO-U2F-Message-Formats]] devices.
+
+    :   <dfn>areCTAPDevicesSupported</dfn>
+    ::  This attribute indicates whether the [=client=] supports communicating with [[FIDO-CTAP]] devices.
+
+    :   <dfn>areResidentKeysSupported</dfn>
+    ::  This attribute indicates whether the [=client=] supports creating [=resident credentials=].
```

Comment:
If you support CTAP devices, isn't this always going to be `true`? Or are clients supposed to guess what users' devices in their pocket can do, and fuzzily adjust depending on what's been paired/plugged in of late?

Reply:
Chrome at the moment (version 75) supports CTAP2 devices, but doesn't have support for account selection etc and so would return `false` for this because resident keys aren't actually usable. I.e. I think these fields are reporting on browser capabilities so it's &ldquo;assuming that the user has an authenticator that does resident keys, will it actually work with this browser?&rdquo;.

Reply:
Resident key support is still being returned as we have a use case for this, and browsers do react differently today.